atom/browser/api/atom_api_app.cc

```diff
@@ -1040,31 +1042,34 @@ Browser::LoginItemSettings App::GetLoginItemSettings(mate::Arguments* args) {
 }
 
 #if defined(USE_NSS_CERTS)
-void App::ImportCertificate(const base::DictionaryValue& options,
-                            const net::CompletionCallback& callback) {
+v8::Local<v8::Promise> App::ImportCertificate(
+    const base::DictionaryValue& options) {
+  scoped_refptr<util::Promise> promise = new util::Promise(isolate());
+
   auto browser_context = AtomBrowserContext::From("", false);
   if (!certificate_manager_model_) {
     auto copy = base::DictionaryValue::From(
         base::Value::ToUniquePtrValue(options.Clone()));
     CertificateManagerModel::Create(
         browser_context.get(),
         base::Bind(&App::OnCertificateManagerModelCreated,
-                   base::Unretained(this), base::Passed(&copy), callback));
-    return;
+                   base::Unretained(this), base::Passed(&copy), promise));
+    return promise->GetHandle();
   }
 
-  int rv = ImportIntoCertStore(certificate_manager_model_.get(), options);
-  callback.Run(rv);
+  promise->Resolve(
+      ImportIntoCertStore(certificate_manager_model_.get(), options));
+  return promise->GetHandle();
 }
 
 void App::OnCertificateManagerModelCreated(
     std::unique_ptr<base::DictionaryValue> options,
-    const net::CompletionCallback& callback,
+    scoped_refptr<util::Promise> promise,
```

Comment:
Should this perhaps be a referance?

Reply:
no, you'll see if you look around that this is how we've implemented all other promisification instances.